File: lua/average.lua
#!/usr/bin/env lua

-- Average or arithmetic mean
-- Andrew Cashner, 2023/11/07

function mean(t)
   local sum = 0
   local count = 0
   
   for k,v in pairs(t) do
       n = tonumber(v)
       if n then
           sum = sum + n
           count = count + 1
       end
   end
  
   return (sum / count)
end

print(mean(arg))

File: javascript/average.js
#!/usr/bin/env node
// Return the mean of a given list of numbers (comma-separated)
// Andrew Cashner, 2024/01/31

function mean(nums) {
  let sum = nums.reduce((a, b) => a + b, 0);
  return sum / nums.length;
}

function main() {
  let input = process.argv.slice(2);
  let nums = input.map(Number);
  if (nums.some(n => isNaN(n))) {
    console.log("Usage: average n1 n2 [n3 n4...] (All arguments must be numbers)");
  } else {
    let avg = mean(nums);
    console.log(avg);
  } 
}

main();

